src/app-layout/__tests__/split-panel-provider.test.tsx

```diff
@@ -0,0 +1,168 @@
+    const reportSize = jest.fn();
+    await renderComponent(
+      <SplitPanelProvider {...defaultProviderProps} size={100} reportSize={reportSize}>
+        <SplitPanel {...defaultSplitPanelProps} />
+      </SplitPanelProvider>
+    );
+
+    expect(reportSize).toHaveBeenCalledTimes(1);
+    expect(reportSize).toHaveBeenLastCalledWith(minSize);
+  });
+
+  test('size cannot be more than maxSize', async () => {
+    const reportSize = jest.fn();
+    await renderComponent(
+      <SplitPanelProvider {...defaultProviderProps} size={800} reportSize={reportSize}>
+        <SplitPanel {...defaultSplitPanelProps} />
+      </SplitPanelProvider>
+    );
+
+    expect(reportSize).toHaveBeenCalledTimes(2);
+    expect(reportSize).toHaveBeenLastCalledWith(500);
+  });
+
+  test('when minSize > maxSize, prefer minSize', async () => {
+    const reportSize = jest.fn();
+    await renderComponent(
+      <SplitPanelProvider
+        {...defaultProviderProps}
+        reportSize={reportSize}
+        getMaxHeight={() => 150}
+        getMaxWidth={() => 150}
+      >
+        <SplitPanel {...defaultSplitPanelProps} />
+      </SplitPanelProvider>
+    );
+
+    expect(reportSize).toHaveBeenCalledTimes(1);
+    expect(reportSize).toHaveBeenCalledWith(minSize);
+  });
+
+  test('onResize handler reports updated size', async () => {
+    const onResize = jest.fn();
+    const { wrapper } = await renderComponent(
+      <SplitPanelProvider {...defaultProviderProps} onResize={onResize}>
+        <SplitPanel {...defaultSplitPanelProps} />
+      </SplitPanelProvider>
+    );
+    resizePanelBy(wrapper, 300);
+    expect(onResize).toHaveBeenCalledWith(300);
+  });
+
+  test('onResize handler size adjusted to min size', async () => {
+    const onResize = jest.fn();
+    const { wrapper } = await renderComponent(
+      <SplitPanelProvider {...defaultProviderProps} onResize={onResize}>
+        <SplitPanel {...defaultSplitPanelProps} />
+      </SplitPanelProvider>
+    );
+    resizePanelBy(wrapper, minSize - 10);
+    expect(onResize).toHaveBeenCalledWith(minSize);
+  });
+
+  test('onResize handler size adjusted to max size', async () => {
+    const onResize = jest.fn();
+    const { wrapper } = await renderComponent(
+      <SplitPanelProvider {...defaultProviderProps} onResize={onResize}>
+        <SplitPanel {...defaultSplitPanelProps} />
+      </SplitPanelProvider>
+    );
+    resizePanelBy(wrapper, 800);
+    expect(onResize).toHaveBeenCalledWith(500);
+  });
+
+  test('onResize handler not called when resize is invalid', async () => {
+    const onResize = jest.fn();
+    const { wrapper } = await renderComponent(
+      <SplitPanelProvider
+        {...defaultProviderProps}
+        onResize={onResize}
+        getMaxHeight={() => 100}
+        getMaxWidth={() => 100}
+      >
+        <SplitPanel {...defaultSplitPanelProps} />
+      </SplitPanelProvider>
+    );
+    resizePanelBy(wrapper, 100);
```

Comment:
Why is this invalid? 100 is not greater than the value returned by `getMaxHeight` or `getMaxWidth`, and the initial size is 0, right?

Reply:
Check is `isResizeValid` definition, https://github.com/cloudscape-design/components/pull/1612/files/d6ac9751cec549e6f5a0fb64343ddcd8dea7f36d?w=1#diff-0f0cfc402450309977133527153e905067bbaa8276200ad22eae14770135290bR32

It does not depend on the resize, we simply prevent `onResize` from firing when it is in confined position. 

I did not introduce this feature, only added a test to confirm